legal-api/src/legal_api/services/filings/validations/incorporation_application.py

```diff
@@ -69,3 +77,67 @@ def validate_offices(incorporation_json) -> Error:
         return msg
 
     return None
+
+
+def validate_roles(incorporation_json) -> Error:
+    """Validate the required completing party of the incorporation filing."""
+    parties_array = incorporation_json['filing']['incorporationApplication']['parties']
+    msg = []
+    completing_party_count = 0
+    incorporator_count = 0
+    director_count = 0
+
+    for item in parties_array:
+        role_array = item['roles']
+
+        if role_array.count('Completing Party') == 1:
+            completing_party_count += 1
+
+        if role_array.count('Incorporator') == 1:
+            incorporator_count += 1
+
+        if role_array.count('Director') == 1:
+            director_count += 1
+
+    if completing_party_count == 0:
+        err_path = '/filing/incorporationApplication/parties/roles'
+        msg.append({'error': 'Must have a minimum of one completing party', 'path': err_path})
+
+    if completing_party_count >= 2:
+        err_path = '/filing/incorporationApplication/parties/roles'
+        msg.append({'error': 'Must have a maximum of one completing party', 'path': err_path})
+
+    if incorporator_count < 1:
```

Comment:
These rules are applicable for BCOMP. For CP it is 3 I believe. The director count for CP is also for 3 I think. Even though we are not implementing CP now, it would be better to have that check around and add a TODO comment or so for CP

Reply:
Yes, your are 100% correct. I was thinking about that but wasn't sure if i should implement something now if we aren't going to use it. I will wrap these conditions in a check tho for BCOMP and add a TODO for CP :) Thanks!

Reply:
Edit* Just added the checks for both types in 👍